.github/workflows/ci_workflows.yml

```diff
@@ -51,14 +51,7 @@ jobs:
       fail-fast: true
       matrix:
         include:
-
-          - name: Code style checks
```

Comment:
Just checking, do the more extensive tests run if pre-commit fails? If not, then we need some conditional logic...

Reply:
Should we turn this into a label triggered CI?

Reply:
I mean, currently the code style check runs quickly relative to the full tests. If the style check fails the other tests stop because of ``fail-fast: true``. Switching pre-commit to the CI bot means these tests don't fail-fast, right? If they do, excellent, if not, then we should make these tests start only if the pre-commit tests succeeds.

Reply:
With the pre-commit.ci bot, I don't see a big need to have all the CI grind to a halt due to failing codestyle checks. The bot can be commanded to autofix the majority of issues that cause codestyle failures anyways. The plan is to have the bot fix any issues right before we merge the PR anyways.

Reply:
Note that we should also mark the pre-commit.ci bot's CI as `Required` so that PRs failing the the style checks do not get merged.

Reply:
SGTM.